### PR TITLE
Add hisat2

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/references/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/references _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core pipelines lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: workflows/references/main.nf [hunk 1]
```diff
@@ -99,6 +103,17 @@ workflow REFERENCES {
         versions = versions.mix(MSISENSORPRO_SCAN.out.versions.first())
     }
 
+    if (tools && tools.split(',').contains('hisat2')) {
+        HISAT2_EXTRACTSPLICESITES(input.gtf)
+        splicesites = input.splice_sites ?: HISAT2_EXTRACTSPLICESITES.out.txt
+
+        HISAT2_BUILD(input.fasta, input.gtf, splicesites)
+
+        hisat2 = HISAT2_BUILD.out.index
```
Comment (sateeshperi) [inline workflows/references/main.nf:112]: do you wanna emit the splicesites out too ?

Comment (sateeshperi) [reply workflows/references/main.nf:112]: or not that imp for downstream ?

**Review by sateeshperi — APPROVED** [review]
lgtm to so far. I would be interested to see how this plugs in to a pipeline. Thanks for this @maxime

File: workflows/references/main.nf (hunk 1 above)
Comment (maxulysse) [reply workflows/references/main.nf:112]: yeah, I'll do later, it's planned

**Pull request merged**